### PR TITLE
Build multi-platform x86 + ARM containers on CI

### PR DESCRIPTION
**Pull request by nilsvu**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

**Comment by knelli2** [comment]
How long will the eumulation take? I'm worried about this taking longer than 6 hours

**Comment by nilsvu** [comment]
Let's wait for https://github.com/nilsvu/spectre/actions/runs/6117406457 :)

File: .github/workflows/BuildDockerContainer.yaml [hunk 1]
```diff
@@ -52,7 +51,7 @@ jobs:
           file: "./containers/Dockerfile.buildenv"
           target: dev
           tags: ${{ inputs.image_name }}:dev
-          platforms: linux/amd64 #,linux/arm64
+          platforms: linux/amd64,linux/arm64
```
Comment (knelli2) [inline .github/workflows/BuildDockerContainer.yaml]: Can we build the container on 2 cores by setting `PARALLEL_MAKE_ARG=-j2` somehow?

Comment (nilsvu) [reply .github/workflows/BuildDockerContainer.yaml]: I think it already builds on 2 cores because that's the default in the container

Comment (knelli2) [reply .github/workflows/BuildDockerContainer.yaml]: Ah yes you are right.

**Comment by nilsvu** [comment]
The ARM container made it to building Charm++, then timed out at 6 hours. https://github.com/sxs-collaboration/spectre/pull/5432 might speed it up sufficiently (it eliminates 2.5 hours just building CMake!). I don't think we have to depend this PR on the speedup PRs since the automatic container deployment isn't critical. And the speedup PRs are easier to test once this is on develop.

**Review by knelli2 — DISMISSED** [review]

**Comment by nilsvu** [comment]
I added a toggle for building the ARM container rather than always building it. This way we can build the ARM container somewhere else if we don't want to wait for the emulation.

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**